doc/source/guide/data_types/timeseries.rst

```diff
@@ -0,0 +1,390 @@
+
+2.1 Creating Custom TimeSeries from a Pandas DataFrame
+-------------------------------------------------------
+
+A TimeSeries object must be supplied with some data when it is
+created.  The data can either be in your current Python session, in a
+local file, or in a remote file.  Let's create some fake data and pass
+it into a TimeSeries object: ::
+
+    >>> import numpy as np
+    >>> intensity = np.sin(np.arange(0, 12 * np.pi, ((12 * np.pi) / (24*60))))
+
+The first line imports the numpy module used to create and store the data.
+The second line creates a basic numpy array of values representing a sine wave.
+We can use this array along with a suitable time storing object (such as AstroPy `~astropy.time` or a list of `datetime` objects) to make a Panda `~pandas.core.frame.DataFrame`.
+A suitable list of times must contain the same number of values as the data, this can be created using: ::
+
+    >>> import datetime
+    >>> base = datetime.datetime.today()
+    >>> times = [base - datetime.timedelta(minutes=x) for x in range(0, 24 * 60)]
+
+The Pandas `~pandas.core.frame.DataFrame` will use the dates list as the index: ::
+
+    >>> from pandas import DataFrame
+    >>> data = DataFrame(intensity, index=times, columns=['intensity'])
+
+This `~pandas.core.frame.DataFrame` can then be used to construct a TimeSeries: ::
+
+    >>> import sunpy.timeseries as ts
+    >>> ts_custom = ts.TimeSeries(data)
+
+Furthermore we could specify the metadata/header and units of this time series by sending them as arguments to the factory: ::
+
+    >>> from collections import OrderedDict
+    >>> import astropy.units as u
+
+    >>> meta = OrderedDict({'key':'value'})
+    >>> units = OrderedDict([('intensity', u.W/u.m**2)])
+    >>> ts_custom = ts.TimeSeries(data, meta, units)
+
+2.2 Creating Custom TimeSeries from an AstroPy Table
+-----------------------------------------------------
+
+A Pandas `~pandas.core.frame.DataFrame` is the underlying object used to store the data within a TimeSeries, so the above example is the most lightweight to create a custom TimeSeries, but being scientific data it will often be more convenient to use an AstroPy `~astropy.table.table.Table` and let the factory convert this.
+An advantage of this method is it allows you to include metadata and AstroPy `~astropy.units.quantity.Quantity` values, which are both supported in tables, without additional arguments.
+For example: ::
+
+    >>> import datetime
+    >>> from astropy.time import Time
+    >>> import astropy.units as u
+    >>> from astropy.table import Table
+
+    >>> base = datetime.datetime.today()
+    >>> times = Time([base - datetime.timedelta(minutes=x) for x in range(0, 24 * 60)])
+    >>> intensity = u.Quantity(np.sin(np.arange(0, 12 * np.pi, ((12 * np.pi) / (24*60)))), u.W/u.m**2)
+    >>> tbl_meta = {'t_key':'t_value'}
+    >>> table = Table([times, intensity], names=['time', 'intensity'], meta=tbl_meta)
+    >>> table.add_index('time')
```

Comment:
I can't replicate this, is sounds like you don't have a table? Because the method add_index(...) is nothing special:
http://docs.astropy.org/en/stable/table/indexing.html
What version of astropy are you using? I'm on V1.3.